hw/sensor/src/sensor.c

```diff
@@ -1162,18 +1184,26 @@ sensor_interrupt_ev_cb(struct os_event *ev)
 static void
 sensor_notify_ev_cb(struct os_event * ev)
 {
-    const struct sensor_notify_ev_ctx *ctx;
+    struct sensor_notify_os_ev *snoe;
     const struct sensor_notifier *notifier;
 
-    ctx = ev->ev_arg;
+    snoe = ev->ev_arg;
+
+    if (!snoe) {
+        return;
+    }
```

Comment:
It doesn't seem like `snoe` can ever be null, but if it does happen, then the event gets leaked here.  The event callback should never return without freeing the event.

Reply:
Actually, snoe is the event, I agree chances of it becoming NULL are slim but it definitely cannot be put back into the pool since, it is NULL and the block address would be invalid then.